app/org/sagebionetworks/bridge/stormpath/StormpathAccount.java

```diff
@@ -171,6 +169,13 @@ public StudyIdentifier getStudyIdentifier() {
         return this.roles;
     }
     @Override
+    public void setRoles(Set<Roles> roles) {
+        this.roles.clear();
+        if (roles != null) {
+            this.roles.addAll(roles);    
+        }
```

Comment:
This can be ImmutableList, let's make sure it is. More important than the role reference being final.